### PR TITLE
bug fixes and simplify inheritance method redefinition

### PR DESCRIPTION
**Pull request by MickSandoz**
-  bug fixes in correspondence_metadata.py
- change a method in import_letters_history.py to simplify inheritance method redefinition

#CO-15

### CODE REVIEW

File: sbc_compassion/models/import_letters_history.py [hunk 1]
```diff
@@ -244,7 +245,7 @@ def _run_analyze(self):
         self.import_completed = True
         logger.info("Imported letters analysis completed.")
 
-    def _analyze_attachment(self, file_data, file_name):
+    def _analyze_attachment(self, file_data, file_name, is_zipfile=False):
```
Comment (ecino) [inline sbc_compassion/models/import_letters_history.py]: `is_zip_file` is never used inside the function, why adding it?

Comment (MickSandoz) [inline sbc_compassion/models/import_letters_history.py]: I know it's my problem, I don't find the way to simplify method redefinition of analyze_attachment in the inherited code (sbc_emai) wihtout add this params.

Without that, redefinition of _run_analyze()(who call analyze_attachment) in sbc_email becomes bigger

Comment (ecino) [inline sbc_compassion/models/import_letters_history.py]: Ok then...

File: sbc_compassion/models/correspondence_metadata.py [hunk 2]
```diff
@@ -42,6 +42,7 @@ def get_correspondence_metadata(self):
         """
         self.ensure_one()
         vals = self.read(self.get_fields())[0]
-        vals['template_id'] = vals['template_id'][0]
+        if vals.get('template_id'):
```
Comment (ecino) [inline sbc_compassion/models/correspondence_metadata.py]: Please refine the test to this:
`if isinstance(vals.get('template_id'), tuple):`

Comment (ecino) [inline sbc_compassion/models/correspondence_metadata.py]: Sorry, you can ignore that. I think the simpler case works fine.

**Pull request merged**